hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectDeltaHistoricalStateCreator.java

```diff
@@ -74,9 +71,9 @@ public void populateHistory() {
         while(ordinal != ORDINAL_NONE) {
             ordinalMapping.put(ordinal, nextOrdinal);
 
-            int shard = ordinal & shardNumberMask;
-            int shardOrdinal = ordinal >> shardOrdinalShift;
-            copyRecord(historicalDataElements, nextOrdinal, stateEngineDataElements[shard], shardOrdinal, currentWriteVarLengthDataPointers);
+            int whichShard = ordinal & shardsHolder.shardNumberMask;
+            int shardOrdinal = ordinal >> shardsHolder.shards[whichShard].shardOrdinalShift;
```

Comment:
I see this `holder.shards[x].shardOrdinalShift` used in a couple of places, I don't think the shard ordinal shift can actually be different between shards -- am I correct?

Reply:
`shardOrdinalShift ` can be different for different shards when we're in the middle of resharding.

Consider the case where we're resharding from 2 shards ```[S0, S1]```  to 4 shards .

In the first step of resharding, we will expand the shard sharray but retain the original shards as is to get-
```[S0, S1, S0, S1]```

In the next step of resharding, we process one shard at a time, so processing S0 yields-
```[S00, S1, S01, S1]```
Here the `dataElements` and `shardOrdinalShift` for the processed shard are different than the original shard but they co exist in the shard holder with shards that havent been processed yet (S1). 

Ultimately when we've finished resharding we arrive at 
```[S00, S10, S01, S11]``` 
and the `shardOrdinalShift`s are once again equal for all shards.